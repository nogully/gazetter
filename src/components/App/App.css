.App {
  text-align: center;
  font-family: 'Merriweather', serif; }

.login {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column; }

h1 {
  font-family: 'UnifrakturCook', cursive; }

form {
  display: flex;
  flex-direction: column;
  width: 30%;
  justify-content: center;
  align-items: center; }

input {
  margin: 20px; }

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px; }

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white; }

.App-title {
  font-size: 4rem; }

.App-intro {
  font-size: large; }

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg); }
  to {
    transform: rotate(360deg); } }
