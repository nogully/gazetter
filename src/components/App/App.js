import React, { Component } from 'react';
import * as api from '../../apiCalls';
import { connect } from 'react-redux';
import './App.css';
import { auth, provider } from '../../firebase';
import { logIn, populateTweets } from '../../actions/actions'
import News from '../News/News.js'
import { Route, withRouter } from 'react-router-dom';
import   today from '../../dateHelper'
// import PropTypes from 'prop-types';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      loading: false
    }
  }

  signIn = () =>  {  
    this.setState( {loading: true} );
    return auth.signInWithPopup(provider)
      .then((user) => {
        this.props.logIn(user);
      }).then(this.fetchTweets())
  }

  fetchTweets = async () => {
    const tweets = await api.getTweets();
    this.props.populateTweets(tweets)
    this.props.history.push('/news')
  }

  render() {
    console.log(this.props.tweets.length)
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Gazetter</h1>
          {this.props.tweets.length? <p id="date">Last updated: { today }</p> : null }
        </header>
        <div className="User">
         { !this.props.tweets.length ? <button onClick={this.signIn}>Sign in with <i className="fab fa-twitter"></i> Twitter</button> : null }
         { this.state.loading && !this.props.tweets.length ? <img src="./newspapers.gif" alt="newspapers" /> : null }
        </div>
        <Route path='/news' component={News} />
      </div>
    );
  }
}
export const mapStateToProps = (store) => ({
  user: store.user,
  tweets: store.tweets
})

export const mapDispatchToProps = (dispatch) => ({
  logIn: user => dispatch(logIn(user)), 
  populateTweets: tweets => dispatch(populateTweets(tweets))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));


