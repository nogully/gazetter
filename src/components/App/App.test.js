/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { auth } from '../../firebase';

describe('App', () => {
  it('should match snapshot', () => { 
    const wrapper = shallow(<App tweets={ [] } user={ {} } logOut={ jest.fn() } populateTweets={ jest.fn() }/>);
    expect(wrapper).toMatchSnapshot();
  })

  describe('signOut', () => {  
    it('should sign out using Firebase', () => {
      const wrapper = shallow(<App tweets={ [] } user={ {} } logOut={ jest.fn() } populateTweets={ jest.fn() }/>);
      auth.signOut = jest.fn()
      wrapper.instance().logOut();
      expect(auth.signOut).toHaveBeenCalled()
    })
  })

  describe('mapStateToProps', () => {
    it('should define the user props for the App container via Redux', () => {
      const mockUser = { display_name: 'noragully' }
      const mockStore = { user: mockUser }
      const expectedProps = { user: { display_name: 'noragully' } }
      const mapped = mapStateToProps(mockStore)
      expect(mapped).toEqual(expectedProps)
    })

    it('should define the tweets props for the App container via Redux', () => {
      const mockTweets = [
              { text: 'Why tho, Jhun?', 
                user: { display_name: 'noragully' } }, 
              { text: 'What\'s good, fam?', 
                user: { display_name: 'yung-jhun' } }
            ]
      const mockStore = { tweets: mockTweets };
      const expectedProps = { tweets: [
              { text: 'Why tho, Jhun?', 
                user: { display_name: 'noragully' } }, 
              { text: 'What\'s good, fam?', 
                user: { display_name: 'yung-jhun' } }
            ] }
      const mapped = mapStateToProps(mockStore);
      expect(mapped).toEqual(expectedProps)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call the Redux dispatch method with logOut', () => {
      const mockDispatch = jest.fn()
      const mapped = mapDispatchToProps(mockDispatch);
      mapped.logOut();
      expect(mockDispatch).toHaveBeenCalled();
    })

    it('should call the Redux dispatch method with populateTweets', () => {
      const mockDispatch = jest.fn()
      const mapped = mapDispatchToProps(mockDispatch);
      mapped.populateTweets();
      expect(mockDispatch).toHaveBeenCalled();
    })
  })
})
