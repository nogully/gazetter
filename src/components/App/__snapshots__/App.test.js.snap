// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    tweets={Array []}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    Gazetter
          </h1>
</header>,
        <div
          className="User"
>
          <button
                    onClick={[Function]}
          >
                    Sign in with 
                    <i
                              className="fab fa-twitter"
                    />
                     Twitter
          </button>
</div>,
        <Route
          component={[Function]}
          path="/news"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="App-title"
>
            Gazetter
</h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gazetter",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Gazetter",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
>
              Sign in with 
              <i
                            className="fab fa-twitter"
              />
               Twitter
</button>,
            null,
          ],
          "className": "User",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Sign in with ",
                <i
                  className="fab fa-twitter"
/>,
                " Twitter",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Sign in with ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fab fa-twitter",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Twitter",
            ],
            "type": "button",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/news",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Gazetter
            </h1>
</header>,
          <div
            className="User"
>
            <button
                        onClick={[Function]}
            >
                        Sign in with 
                        <i
                                    className="fab fa-twitter"
                        />
                         Twitter
            </button>
</div>,
          <Route
            component={[Function]}
            path="/news"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              Gazetter
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Gazetter",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Gazetter",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
>
                Sign in with 
                <i
                                className="fab fa-twitter"
                />
                 Twitter
</button>,
              null,
            ],
            "className": "User",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Sign in with ",
                  <i
                    className="fab fa-twitter"
/>,
                  " Twitter",
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Sign in with ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fab fa-twitter",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Twitter",
              ],
              "type": "button",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/news",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
